sycl/doc/DeviceGlobal.md

```diff
@@ -0,0 +1,658 @@
+  content of the variable is stored in this buffer.  All device images on a
+  particular device point to the same buffer, so the variable's state is
+  shared.  The runtime, therefore, must allocate this USM buffer for each such
+  device global variable.
+
+* As we noted above, the front-end generates new content in the integration
+  footer which calls the function `sycl::detail::device_global_map::add()`.
+  The runtime defines this function and maintains information about all the
+  device global variables in the application.  This information includes:
+
+  - The host address of the variable.
+  - The string which uniquely identifies the variable.
+  - The size (in bytes) of the underlying `T` type for the variable.
+  - A boolean telling whether the variable is decorated with the
+    `device_image_scope` property.
+  - The associated per-device USM buffer pointer, if this variable does not
+    have the `device_image_scope` property.
+
+  We refer to this information as the "device global database" below.
+
+* The runtime also implements the new `copy` and `memcpy` functions in the
+  `queue` and `handler` classes which copy to or from `device_global`
+  variables.
+
+#### Initializing the device global variables in device code
+
+When a DPC++ application submits a kernel, the runtime constructs a
+`pi_program` containing this kernel that is compiled for the target device, if
+such a `pi_program` does not yet exist.  If the kernel resides in a device code
+module that calls into a shared library, the runtime identifies a set of device
+code modules that need to be online-linked together in order to construct the
+`pi_program`.
+
+After creating a `pi_program` and before invoking any kernel it contains, the
+runtime does the following:
+
+* Scan the strings in the "device-globals" properties of the
+  "SYCL/misc properties" sets of each device code module that contributes to
+  the `pi_program` to get the unique string associated with each device global
+  variable that is used by the `pi_program`.  For each of these strings, the
+  runtime uses the device global database to see if the variable was decorated
+  with `device_image_scope`.  If it was not so decorated and if a USM buffer
+  has not already been created for the variable on this target device, the
+  runtime allocates the buffer from USM device memory using the size from the
+  database and zero-initializes the content of the buffer.  The pointer to this
+  buffer is saved in the database for future reuse.
+
+* For each device global variable that is not decorated with
+  `device_image_scope`, the runtime initializes the `usmptr` member in the
+  *device instance* of the variable by using a backend-specific function which
+  copies data from the host to a device variable.  It is a simple matter to use
+  this function to overwrite the `usmptr` member with the address of the USM
+  buffer.  The details of this device-specific function are described below.
+
```

Comment:
Can we have a couple of words saying how we initialize a global variable when `device_image_scope` is passed?

Reply:
Thanks.

Reply:
I did this in da1124837b604f6941e0807ece7cd2292e66f136.